include/alpaka/mem/view/ViewAccessor.hpp

```diff
@@ -0,0 +1,302 @@
+            inline
+#else
+            static
+#endif
+                constexpr bool isView<
+                    TView,
+                    meta::Void<
+                        Idx<TView>,
+                        Dim<TView>,
+                        decltype(alpaka::getPtrNative(std::declval<TView>())),
+                        decltype(getPitchBytes<0>(std::declval<TView>())),
+                        decltype(extent::getExtent<0>(std::declval<TView>()))>> = true;
+
+            template<typename... TAccessModes>
+            struct BuildAccessModeList;
+
+            template<typename TAccessMode>
+            struct BuildAccessModeList<TAccessMode>
+            {
+                using type = TAccessMode;
+            };
+
+            template<typename TAccessMode1, typename TAccessMode2, typename... TAccessModes>
+            struct BuildAccessModeList<TAccessMode1, TAccessMode2, TAccessModes...>
+            {
+                using type = std::tuple<TAccessMode1, TAccessMode2, TAccessModes...>;
+            };
+
+            ALPAKA_NO_HOST_ACC_WARNING
+            template<
+                typename... TAccessModes,
+                typename TViewForwardRef,
+                std::size_t... TPitchIs,
+                std::size_t... TExtentIs>
+            ALPAKA_FN_HOST_ACC auto buildViewAccessor(
+                TViewForwardRef&& view,
+                std::index_sequence<TPitchIs...>,
+                std::index_sequence<TExtentIs...>)
+            {
+                using TView = std::decay_t<TViewForwardRef>;
+                static_assert(isView<TView>, "");
+                using TBufferIdx = Idx<TView>;
+                constexpr auto dim = Dim<TView>::value;
+                using Elem = Elem<TView>;
+                auto p = getPtrNative(view);
+                static_assert(
+                    std::is_same<decltype(p), const Elem*>::value || std::is_same<decltype(p), Elem*>::value,
+                    "We assume that getPtrNative() returns a raw pointer to the view's elements");
+                static_assert(
+                    !std::is_same<decltype(p), const Elem*>::value
+                        || std::is_same<std::tuple<TAccessModes...>, std::tuple<alpaka::ReadAccess>>::value,
+                    "When getPtrNative() returns a const raw pointer, the access mode must be ReadAccess");
+                using AccessModeList = typename BuildAccessModeList<TAccessModes...>::type;
+                return Accessor<Elem*, Elem, TBufferIdx, dim, AccessModeList>{
+                    const_cast<Elem*>(p), // strip constness, this is handled the the access modes
+                    getPitchBytes<TPitchIs + 1>(view)...,
+                    {extent::getExtent<TExtentIs>(view)...}};
+            }
+        } // namespace internal
+
+        //! Builds an accessor from view like memory objects.
+        template<typename TView>
+        struct BuildAccessor<TView, std::enable_if_t<internal::isView<TView>>>
```

Comment:
This is still very broad; all buffer implementations (including my `BufGenericSycl`) fulfill the requirement `isView`. This leads to a compilation error when trying to specialize `BuildAccessor`:

```
/home/stepha27/workspace/zoo/alpaka/include/alpaka/mem/buf/BufGenericSycl.hpp:331:16: note: partial specialization matches [with TElem = float, TDim = std::integral_constant<unsigned long, 1>, TIdx = unsigned long, TDev = alpaka::DevGenericSycl<alpaka::PltfCpuSyclIntel>]
        struct BuildAccessor<BufGenericSycl<TElem, TDim, TIdx, TDev>>
               ^
/home/stepha27/workspace/zoo/alpaka/include/alpaka/mem/view/ViewAccessor.hpp:289:16: note: partial specialization matches [with TView = alpaka::BufGenericSycl<float, std::integral_constant<unsigned long, 1>, unsigned long, alpaka::DevGenericSycl<alpaka::PltfCpuSyclIntel>>]
        struct BuildAccessor<TView, std::enable_if_t<internal::isView<TView>>>
```

Reply:
What is the return type of `getPtrNative()` on a SYCL buffer? If it is not a `T*`, then I can distinguish by this trait.